### PR TITLE
Simplify pgp util exceptions

### PR DESCRIPTION
**Pull request by rkondratowicz**
Throw runtime exceptions defined in the encryption module from the util class instead of forcing the clients to handle ambiguous `IOException` and bouncycastle's exception.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java [hunk 1]
```diff
@@ -44,43 +48,49 @@ private PgpEncryptionUtil() {
...
-            );
-
-        PGPEncryptedDataGenerator encryptedDataGenerator =
-            prepareDataEncryptor(
-                pgpPublicKey,
-                withIntegrityCheck
-            );
-
-        return writeEncryptedDataToOutputStream(byteArrayOutputStream, encryptedDataGenerator);
+    ) {
```
Comment (rkondratowicz) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java]: best viewed by adding `?w=1` to url

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java [hunk 2]
```diff
@@ -44,43 +48,49 @@ private PgpEncryptionUtil() {
...
-                getDecoderStream(new ByteArrayInputStream(data))
-            )
-        ).orElseThrow(() -> new UnableToLoadPgpPublicKeyException("PGP Public key object could not be constructed"));
+    public static PGPPublicKey loadPublicKey(byte[] data) {
+        try {
+            return lookupPublicSubkey(
+                new BcPGPPublicKeyRing(
+                    getDecoderStream(new ByteArrayInputStream(data))
+                )
+            ).orElseThrow(() -> new UnableToLoadPgpPublicKeyException(null));
```
Comment (banderous) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java]: Unrelated to this change, but should UnableToLoadPgpPublicKeyException exist as an Exception? It seems more like an assertion failure that should immediately crash the app, it isn't an exceptional condition that can be handled.

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java [hunk 3]
```diff
@@ -44,43 +48,49 @@ private PgpEncryptionUtil() {
...
+            PGPEncryptedDataGenerator encryptedDataGenerator =
+                prepareDataEncryptor(
+                    pgpPublicKey,
+                    withIntegrityCheck
+                );
+
+            return writeEncryptedDataToOutputStream(byteArrayOutputStream, encryptedDataGenerator);
+        } catch (IOException | PGPException exc) {
+            log.error("Error encrypting file", exc);
+            throw new UnableToPgpEncryptZipFileException(exc);
```
Comment (doncem) [inline src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java]: I would go for more abstract name, e.g. `UnableToEncryptFileException`. There is no need to be extremely verbose with exception names

File: src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java (hunk 2 above)
Comment (doncem) [reply src/main/java/uk/gov/hmcts/reform/sendletter/services/encryption/PgpEncryptionUtil.java]: Same here, could do with just `UnableToLoadPublicKeyException`. Not even pgp related then. specifically for any attempt to load public key

**Pull request merged**